test/dhtest.c

```diff
@@ -535,10 +549,28 @@ static int run_rfc5114_tests(void)
     }
     return 1;
  bad_err:
+    if (dhA)
+        DH_free(dhA);
+    if (dhB)
+        DH_free(dhB);
+    if (Z1)
+        OPENSSL_free(Z1);
+    if (Z2)
+        OPENSSL_free(Z2);
+
```

Comment:
Checking for NULL before using the freeing routines isn't (or shouldn't be) needed.